node-graph/gcore/src/vector/vector_nodes.rs

```diff
@@ -1138,6 +1138,97 @@ where
 	output_table
 }
 
+/// Simplifies a polyline using the Ramer–Douglas–Peucker algorithm.
+#[node_macro::node(category("Vector: Modifier"), name("Decimate"), path(graphene_core::vector))]
+async fn decimate(
+	_: impl Ctx,
+	vector_data: VectorDataTable,
+	#[default(1.0)]
+	#[hard_min(0.0)]
+	tolerance: f64,
+) -> VectorDataTable {
+	let mut result_table = VectorDataTable::default();
+
+	for vector_data_instance in vector_data.instance_iter() {
+		let mut result = VectorData {
+			style: vector_data_instance.instance.style.clone(),
+			..Default::default()
+		};
+
+		for subpath in vector_data_instance.instance.stroke_bezier_paths() {
+			let points: Vec<DVec2> = subpath.manipulator_groups().iter().map(|g| g.anchor).collect();
+			let closed = subpath.closed();
+
+			let simplified = if points.len() > 2 { douglas_peucker(&points, tolerance, closed) } else { points };
+
+			// Rebuild the subpath from the simplified points
+			let mut new_groups = Vec::new();
+			let mut id_gen = PointId::generate();
+			for pt in simplified {
+				new_groups.push(bezier_rs::ManipulatorGroup {
+					anchor: pt,
+					in_handle: None,
+					out_handle: None,
+					id: id_gen,
+				});
+				id_gen = id_gen.next_id();
+			}
+			let new_subpath = Subpath::new(new_groups, closed);
+			result.append_subpath(new_subpath, closed);
+		}
+
+		result_table.push(Instance {
+			instance: result,
+			transform: vector_data_instance.transform,
+			alpha_blending: vector_data_instance.alpha_blending,
+			source_node_id: vector_data_instance.source_node_id,
+		});
+	}
+
+	result_table
+}
+
+fn douglas_peucker(points: &[DVec2], epsilon: f64, closed: bool) -> Vec<DVec2> {
+	if points.len() < 2 {
+		return points.to_vec();
+	}
+
+	let (_first, _lastt) = (0, points.len() - 1);
+
+	// For closed paths, treat as open for simplification, then close at the end
+	let (_first, _last, _is_closed) = if closed && points.len() > 2 { (0, points.len() - 1, true) } else { (0, points.len() - 1, false) };
```

Comment:
Line could be simplified to
```rs
let is_closed = closed && points.len() > 2;
```
Since the other variables don't change and were already defined.